### PR TITLE
Adjust catalog to work with luigified console

### PR DESCRIPTION
**Pull request by y-kkamil**
**Description**

Changes proposed in this pull request:

Adjusts catalog view to work in luigified console

Resolves #296

### CODE REVIEW

File: catalog/src/commons/builder.js [hunk 1]
```diff
@@ -13,11 +13,9 @@ class Builder {
       }
       const timeout = setTimeout(resolve, 1000);
       window.addEventListener('message', e => {
```
Comment (dariadomagala) [inline catalog/src/commons/builder.js]: Why not using LuigiClient here?

File: catalog/src/commons/builder.js [hunk 2]
```diff
@@ -13,11 +13,9 @@ class Builder {
...
-        if (!e.data || e.data[0] !== 'init') return;
-        const data = e.data[1];
-        this.currentEnvironmentId = data.currentEnvironmentId;
-        this.token = data.idToken;
-        this.sessionId = data.sessionId;
+        if (!e.data || !e.data.context || e.data.msg !== 'luigi.init') return;
+        this.currentEnvironmentId = e.data.context.environmentId;
```
Comment (dariadomagala) [inline catalog/src/commons/builder.js]: In this approach `context` is a stringified JSON, so you would need to parse it first. Switching to LuigiClient would solve that tho.

File: core/src/assets/config/console-config.js [hunk 3]
```diff
@@ -18,12 +18,24 @@ function getNodes(environment) {
     },
     {
       category: 'Service Catalog',
```
Comment (dariadomagala) [inline core/src/assets/config/console-config.js]: Switch to LuigiClient in catalog view, and pass the context data (environment and token) here.

File: catalog/src/commons/builder.js [hunk 4]
```diff
@@ -13,11 +13,9 @@ class Builder {
       }
       const timeout = setTimeout(resolve, 1000);
       window.addEventListener('message', e => {
-        if (!e.data || e.data[0] !== 'init') return;
-        const data = e.data[1];
-        this.currentEnvironmentId = data.currentEnvironmentId;
-        this.token = data.idToken;
-        this.sessionId = data.sessionId;
```
[the remark this answers is not in the document]
Comment (y-kkamil) [reply catalog/src/commons/builder.js]: removed

File: catalog/src/commons/builder.js (hunk 1 above)
Comment (y-kkamil) [reply catalog/src/commons/builder.js]: i am asking myself the same question.. why? ;) fixed

File: catalog/src/commons/builder.js (hunk 2 above)
Comment (y-kkamil) [reply catalog/src/commons/builder.js]: fixed

**Comment by y-kkamil** [comment]
another issue i have just spotted - after creating new instance, overlay is removed only from the content view and not from the luigi frame - somehow `handleCloseModal` handler is not being fired

File: catalog/src/commons/builder.js [hunk 5]
```diff
@@ -28,16 +27,12 @@ class Builder {
     if (!this.token) {
       return null;
     }
-    return `Bearer ${this.token}`;
+    return this.token;
```
Comment (dariadomagala) [inline catalog/src/commons/builder.js]: Wouldn't that be enough to leave just `return this.token`? Since you want to return null when it's null :)

Comment (y-kkamil) [reply catalog/src/commons/builder.js]: fixed

**Pull request merged**